stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
     }
+
+    // [time.zone.leap]
+
+    class leap_second {
+    public:
+        leap_second(const leap_second&) = default;
+        leap_second& operator=(const leap_second&) = default;
+
+        constexpr leap_second(const sys_seconds& _Date_, bool _Is_positive_) noexcept
+            : _Date{_Date_}, _Is_positive{_Is_positive_} {}
+
+        _NODISCARD constexpr sys_seconds date() const noexcept {
+            return _Date;
+        }
+
+        _NODISCARD constexpr seconds value() const noexcept {
+            return _Is_positive ? seconds{1} : seconds{-1};
+        }
+
+        _NODISCARD constexpr bool _Positive() const noexcept {
+            return _Is_positive;
+        }
+
+    private:
+        sys_seconds _Date;
+        bool _Is_positive;
+    };
+
+    _NODISCARD inline bool operator==(const leap_second& _Left, const leap_second& _Right) noexcept {
+        return _Left.date() == _Right.date();
+    }
+    template <class _Duration>
+    _NODISCARD bool operator==(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() == _Right;
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator<(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() < _Right;
+    }
+    template <class _Duration>
+    _NODISCARD bool operator<(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return _Left < _Right.date();
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator>(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Right < _Left.date();
+    }
+    template <class _Duration>
+    _NODISCARD bool operator>(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return _Right.date() < _Left;
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator<=(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return !(_Right < _Left.date());
+    }
+    template <class _Duration>
+    _NODISCARD bool operator<=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Right.date() < _Left);
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator>=(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return !(_Left.date() < _Right);
+    }
+    template <class _Duration>
+    _NODISCARD bool operator>=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Left < _Right.date());
+    }
+
+#ifdef __cpp_lib_concepts
+    template <class _Duration>
+    requires three_way_comparable_with<sys_seconds, sys_time<_Duration>> constexpr auto operator<=>(
+        const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() <=> _Right;
+    }
+    constexpr inline strong_ordering operator<=>(const leap_second& _Left, const leap_second& _Right) noexcept {

```

Comment:
Should be `_NODISCARD`. Also, `constexpr` implies `inline` for functions, so we conventionally avoid repeating `inline`.

WG21-N4878 [dcl.constexpr]/1: "A function or static data member declared with the `constexpr` or `consteval` specifier is implicitly an inline function or variable (9.2.8)."